app/scripts/lib/transaction/smart-transactions.ts

```diff
@@ -251,40 +251,44 @@ class SmartTransactionHook {
   }
 
   async #addListenerToUpdateStatusPage({ uuid }: { uuid: string }) {
-    this.#smartTransactionsController.eventEmitter.on(
-      `${uuid}:smartTransaction`,
+    this.#controllerMessenger.subscribe(
+      'SmartTransactionsController:smartTransaction',
       async (smartTransaction: SmartTransaction) => {
-        const { status } = smartTransaction;
-        if (!status || status === SmartTransactionStatuses.PENDING) {
-          return;
-        }
-        if (!this.#approvalFlowEnded) {
-          await this.#updateApprovalRequest({
-            smartTransaction,
-          });
+        if (smartTransaction.uuid === uuid) {
```

Comment:
Yes, as you can see, we previously subscribed to `${uuid}:smartTransaction` using an event emitter. However, the controller now only publishes `SmartTransactionsController:smartTransaction`. This means it's the consumer's responsibility to filter and handle events related to their specific UUID. Do you think it would be helpful to include this as a comment in the code?